src/blob/index/simple.rs

```diff
@@ -462,7 +502,7 @@ impl Index for Simple {
 
     async fn count(&self) -> Result<usize> {
         match &self.inner {
-            State::InMemory(headers) => Ok(headers.len()),
+            State::InMemory(headers) => Ok(headers.values().fold(0, |acc, x| acc + x.len())),
```

Comment:
I think `IndexHeader` should have records count.

Reply:
Done